apps/prairielearn/src/lib/assessment.ts

```diff
@@ -213,22 +208,15 @@ export async function gradeAssessmentInstanceAsync(
   await async.eachSeries(variants, async (row) => {
     debug('gradeAssessmentInstance()', 'loop', 'variant.id:', row.variant.id);
     const check_submission_id = null;
-    const gradingJobId = await promisify(gradeVariant)(
+    await gradeVariantAsync(
       row.variant,
       check_submission_id,
       row.question,
       row.variant_course,
       authn_user_id,
       overrideGradeRate,
     );
-    if (gradingJobId !== undefined) {
-      externalGradingJobIds.push(gradingJobId);
-    }
   });
-  if (externalGradingJobIds.length > 0) {
-    // We need to submit these grading jobs to be graded
-    await externalGrader.beginGradingJobs(externalGradingJobIds);
-  }
```

Comment:
The `gradeVariant` function did not return a grading job in its previous form, nor does it do that in the current form, so this code was unnecessary. The grading jobs are started in `gradeVariant` itself.